extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/devmode/DevConsoleCORSFilter.java

```diff
@@ -0,0 +1,51 @@
+package io.quarkus.vertx.http.runtime.devmode;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.jboss.logging.Logger;
+
+import io.quarkus.vertx.http.runtime.cors.CORSConfig;
+import io.quarkus.vertx.http.runtime.cors.CORSFilter;
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.ext.web.RoutingContext;
+
+public class DevConsoleCORSFilter implements Handler<RoutingContext> {
+    private static final Logger LOG = Logger.getLogger(DevConsoleCORSFilter.class);
+
+    private static final String HTTP_PORT_CONFIG_PROP = "quarkus.http.port";
+    private static final String LOCAL_HOST = "localhost";
+    private static final String HTTP_LOCAL_HOST = "http://" + LOCAL_HOST;
+    private static final String HTTPS_LOCAL_HOST = "https://" + LOCAL_HOST;
+
+    public DevConsoleCORSFilter() {
+    }
+
+    private static CORSFilter corsFilter() {
+        int httpPort = ConfigProvider.getConfig().getValue(HTTP_PORT_CONFIG_PROP, int.class);
+        CORSConfig config = new CORSConfig();
+        config.origins = Optional.of(List.of(HTTP_LOCAL_HOST + ":" + httpPort, HTTPS_LOCAL_HOST + ":" + httpPort));
+        return new CORSFilter(config);
+    }
+
+    @Override
+    public void handle(RoutingContext event) {
+        HttpServerRequest request = event.request();
+        HttpServerResponse response = event.response();
+        String origin = request.getHeader(HttpHeaders.ORIGIN);
+        if (origin == null) {
+            corsFilter().handle(event);
+        } else {
+            if (origin.startsWith(HTTP_LOCAL_HOST) || origin.startsWith(HTTPS_LOCAL_HOST)) {
```

Comment:
Will this check work with IPv6 addresses?

I guess it assumes `::1 localhost` entry in `/etc/hosts`

Reply:
Hi @rsvoboda, The goal of the handler  is not check all the possible variations, we don't really expect users enter IPv6 addresses in the browser :-) when working with DevUI, we can tune things as needed going forward

Reply:
@rsvoboda Sorry, my last comment was a bit confusing, note, the typical mode of operation won't be affected, if users will want to enter IPv6 addresses, fine, since DevUI pages are not using scripts, it all will work, no `Origin` will be sent and this filter will just let the request through.

To make things predictable/simple, we make sure that if someone writes a DevUI pages using Fetch or XHR API, then it will work only if it will run on localhost. Non local host requests are rejected